pkg/controllers/hub/trafficmanagerprofile/controller.go

```diff
@@ -226,41 +226,117 @@ func (r *Reconciler) handleUpdate(ctx context.Context, profile *fleetnetv1beta1.
 		return false
 	}
 
-	if current.Tags == nil {
+	// Compare tags
+	if !equalTags(current.Tags, desired.Tags) {
+		return false
+	}
+
+	return true
+}
+
+// hasRequiredProperties checks if the profile has all required properties
+func hasRequiredProperties(profile armtrafficmanager.Profile) bool {
+	return profile.Properties != nil &&
+		profile.Properties.MonitorConfig != nil &&
+		profile.Properties.ProfileStatus != nil &&
+		profile.Properties.TrafficRoutingMethod != nil &&
+		profile.Properties.DNSConfig != nil
+}
+
+// equalMonitorConfig compares the monitoring configuration
+func equalMonitorConfig(current, desired *armtrafficmanager.MonitorConfig) bool {
+	if current.IntervalInSeconds == nil || current.Path == nil ||
+		current.Port == nil || current.Protocol == nil ||
+		current.TimeoutInSeconds == nil || current.ToleratedNumberOfFailures == nil {
+		return false
+	}
+
+	return *current.IntervalInSeconds == *desired.IntervalInSeconds &&
+		*current.Path == *desired.Path &&
+		*current.Port == *desired.Port &&
+		*current.Protocol == *desired.Protocol &&
+		*current.TimeoutInSeconds == *desired.TimeoutInSeconds &&
+		*current.ToleratedNumberOfFailures == *desired.ToleratedNumberOfFailures
+}
+
+// equalCustomHeaders compares custom headers between profiles
+func equalCustomHeaders(currHeaders, desiredHeaders []*armtrafficmanager.MonitorConfigCustomHeadersItem) bool {
+	// If lengths are different, headers are not equal
+	if len(currHeaders) != len(desiredHeaders) {
 		return false
 	}
 
-	for key, value := range desired.Tags {
-		currentValue := current.Tags[key]
-		if (value == nil && currentValue != nil) || (value != nil && currentValue == nil) || (currentValue == nil || *currentValue != *value) {
+	// Create a map of current headers for easier comparison
+	currHeadersMap := make(map[string]string)
+	for _, header := range currHeaders {
+		if header.Name == nil || header.Value == nil {
+			return false
+		}
+		currHeadersMap[*header.Name] = *header.Value
+	}
+
+	// Check if all desired headers are in current headers with same values
+	for _, header := range desiredHeaders {
+		if header.Name == nil || header.Value == nil {
+			continue // Skip nil values in desired (shouldn't happen but just in case)
+		}
+
+		val, exists := currHeadersMap[*header.Name]
+		if !exists || val != *header.Value {
+			return false
+		}
+	}
+
+	return true
+}
+
+// equalProfileProperties compares profile status and traffic routing method
+func equalProfileProperties(current, desired *armtrafficmanager.ProfileProperties) bool {
+	return *current.ProfileStatus == *desired.ProfileStatus &&
+		*current.TrafficRoutingMethod == *desired.TrafficRoutingMethod
+}
+
+// equalDNSConfig compares DNS configuration
+func equalDNSConfig(current, desired *armtrafficmanager.DNSConfig) bool {
+	return current.TTL != nil && *current.TTL == *desired.TTL
+}
+
+// equalTags compares tags between profiles
+func equalTags(currentTags, desiredTags map[string]*string) bool {
```

Comment:
name is not accurate, the func is to find whether the desiredTags has been in currentTags.